### ISSUE TITLE
[16.04] Remove U1DB

### ISSUE THREAD

**Issue by UniversalSuperBox** [ubports/ubports-touch#96]
A decision was made in [our latest developer meeting](https://forums.ubports.com/topic/273/june-1-2017-app-developer-os-developer-meeting/2) to get rid of U1DB in the core Ubuntu image.

U1DB is a database meant to store settings and data for Ubuntu apps. It was meant to sync using the Ubuntu One platform. Since Canonical is no longer supporting Ubuntu for Devices and U1DB is reportedly complicated and inconsistent, it will be removed from Ubuntu Touch 16.04.

**Comment by bastos77** [comment]
But what would be the alternative to store settings and data for Ubuntu apps in 16.04?

**Comment by UniversalSuperBox** [comment]
An app should move toward using Qsettings to store settings and QML LocalStorage for data storage.

**Comment by NeoTheThird** [comment]
@bastos77 it's not difficult to migrate and it gives us better compatibility and less code to maintain.

**Comment by dobey** [comment]
I would say you should use LocalStorage for settings too. QSettings currently has an issue with how it writes data, which can lead to issues with reading the settings again, when storage is near full.

**Comment by NeoTheThird** [comment]
But QSettings is still in development upstream, so issues at least have a chance of getting fixed. Supporting it makes sense, but we don't have to encourage using it.

**Comment by dark-eye** [comment]
Oh no , I just read the LocalStorage documentation and it`s SQL based...
Please don't  replace U1DB until we have a good document based DB solution as SQL create quite a lot of overhead in development.

I personally would  prefer to never use SQL in my life as it a very bad solution for none object connection oriented programs (which are most of the apps)

**Comment by Flohack74** [comment]
@dark-eye A big part of my work life consists of using SQL, and I can tell you its not that bad, even for loosely-related data :) - But, I can understand the idea. Nevertheless, have you goo ideas for alternatives? Can you imagine using smth like this? http://samhogy.co.uk/2017/07/connecting-sqlite-to-graphql.html

**Comment by Flohack74** [comment]
Or this https://unqlite.org/

**Comment by dark-eye** [comment]
Hey @Flohack74 I`m actually working with SQL in my work too ( billing for telecom :.( ), the unqlite looks cool maybe we can (would glad to help if need ) add a QML component to interface with it as most of the app developers will try and stay away from C++ if they don't have to ( or is it just me ? :) )

**Comment by Flohack74** [comment]
I am voting to remove this issue from OTA-4. We will not be ready by end of June with having a replacement in place AND all Apps ported to it.

**Comment by NeoTheThird** [comment]
Right, but let's keep it on the kanban board, since work on that can already start.

**Comment by dpniel** [comment]
Have we listed anywhere the actual issues with U1DB could we not improve on what was previously done? U1DB is a sqlite database under the hood so if the issues are just around the querying/sorting maybe it wouldn't be too much effort to get it into shape?

**Comment by NeoTheThird** [comment]
The main issue is that it does not offer anything that upstream solutions don't. That reduces portability and adds work for us.
